olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp

```diff
@@ -152,6 +153,35 @@ CatalogVersionResponse CatalogRepository::GetLatestVersion(
   return version_response;
 }
 
+VersionsResponse CatalogRepository::GetVersionsList(
+    const client::HRN& catalog,
+    client::CancellationContext cancellation_context,
+    const VersionsRequest& request, const client::OlpClientSettings& settings) {
+  auto fetch_option = request.GetFetchOption();
+  repository::CatalogCacheRepository repository(catalog, settings.cache);
+  if (fetch_option == CacheOnly) {
```

Comment:
What about CacheWithUpdate option?

Reply:
It is not clear for now how to cache requests. So it is temporary will request online.

Reply:
CacheWithUpdate does not make much sense for versioned catalogs as you always get the same result back. It is more for volatile layers as there the value can change every time you request.